pkg/ansible/proxy/proxy.go

```diff
@@ -243,15 +251,31 @@ func addWatchToController(owner kubeconfig.NamespacedOwnerReference, cMap *contr
 			awMap.Store(resource.GroupVersionKind())
 			typeString := fmt.Sprintf("%v.%v", owner.Kind, ownerGV.Group)
 			log.Info("Watching child resource", "kind", resource.GroupVersionKind(), "enqueue_annotation_type", typeString)
-			err = contents.Controller.Watch(&source.Kind{Type: resource}, &osdkHandler.EnqueueRequestForAnnotation{Type: typeString}, dependentPredicate)
+			err = addWatch(contents.Controller, &source.Kind{Type: resource}, &osdkHandler.EnqueueRequestForAnnotation{Type: typeString}, dependentPredicate)
 			if err != nil {
+				log.Error(err, "GVK", resource.GroupVersionKind())
 				return err
 			}
 		}
 	}
 	return nil
 }
 
+func addWatch(c controller.Controller, s source.Source, eh handler.EventHandler, predicates ...predicate.Predicate) error {
+	errChan := make(chan error, 1)
+	go func() {
+		err := c.Watch(s, eh, predicates...)
+		errChan <- err
```

Comment:
> We should work upstream to get a controller.WatchWithContext() function so that we don't need to resort to this.
> The problem here is that it is not possible to cancel the c.Watch() call, so if we reach the deadline, the c.Watch call will just stay running in the background forever. This is effectively a resource leak.

Reply:
required to address.

Reply:
It shows no longer required.